packages/components/src/HeaderBar/HeaderBar.component.js

```diff
@@ -214,44 +211,40 @@ function Products({ renderers, t, ...props }) {
 }
 
 function HeaderBar(props) {
-	const renderers = getRenderers(props.renderers);
-	const Components = Object.assign(
-		{
-			Logo,
-			Brand,
-			Environment,
-			Search,
-			User,
-			Information,
-			Help,
-			Products,
-			AppNotification,
-		},
-		props.renderers || {},
-	);
+	const Components = Inject.getAll(props.getComponent, {
+		Logo,
+		Brand,
+		Environment,
+		Search,
+		User,
+		Information,
+		Help,
+		Products,
+		AppNotification,
+	});
 
 	return (
 		<nav className={classNames(theme['tc-header-bar'], 'tc-header-bar', 'navbar')}>
 			<ul className={theme['tc-header-bar-actions']}>
-				{props.logo && <Components.Logo renderers={renderers} {...props.logo} t={props.t} />}
+				{props.logo && <Components.Logo getComponent={props.getComponent} {...props.logo} t={props.t} />}
 				{props.brand && (
-					<Components.Brand renderers={renderers} {...props.brand} isSeparated={!!props.env} />
+					<Components.Brand getComponent={props.getComponent} {...props.brand} isSeparated={!!props.env} />
 				)}
-				{props.env && <Components.Environment renderers={renderers} {...props.env} />}
+				{props.env && <Components.Environment getComponent={props.getComponent} {...props.env} />}
 			</ul>
 			<ul className={classNames(theme['tc-header-bar-actions'], theme.right)}>
-				{props.search && <Components.Search renderers={renderers} {...props.search} />}
+				{props.search && <Components.Search getComponent={props.getComponent} {...props.search} />}
 				{props.notification && (
-					<Components.AppNotification renderers={renderers} {...props.notification} t={props.t} />
+					<Components.AppNotification getComponent={props.getComponent} {...props.notification} t={props.t} />
 				)}
-				{props.help && <Components.Help renderers={renderers} {...props.help} t={props.t} />}
+				{props.help && <Components.Help getComponent={props.getComponent} {...props.help} t={props.t} />}
 				{!props.user &&
 					props.information && (
-						<Components.Information renderers={renderers} {...props.information} t={props.t} />
+						<Components.Information getComponent={props.getComponent} {...props.information} t={props.t} />
 					)}
-				{props.user && <Components.User renderers={renderers} {...props.user} />}
+				{props.user && <Components.User getComponent={props.getComponent} {...props.user} />}
 				{props.products && (
-					<Components.Products renderers={renderers} {...props.products} t={props.t} />
+					<Components.Products getComponent={props.getComponent} {...props.products} t={props.t} />
```

Comment:
For this block of code, don't we want to have some placeholder to inject others components ?